File: app/backend/src/application/commands/models/create_order_command.rs
use serde::{Deserialize, Serialize};

/// 注文作成用のアイテム
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateOrderCommandItem {
    pub sku_id: String,
    pub quantity: u32,
}

/// 注文作成用の顧客情報
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateOrderCommandCustomerInfo {
    pub first_name: String,
    pub last_name: String,
    pub email: String,
    pub phone: String,
}

/// 注文作成用の配送先情報
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateOrderCommandShippingAddress {
    pub postal_code: String,
    pub prefecture: String,
    pub city: String,
    pub street_address: String,
    pub building: Option<String>,
}

/// 注文作成コマンド
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateOrderCommand {
    pub customer_info: CreateOrderCommandCustomerInfo,
    pub items: Vec<CreateOrderCommandItem>,
    pub shipping_method_id: String,
    pub payment_method_id: String,
    pub shipping_address: CreateOrderCommandShippingAddress,
}

impl CreateOrderCommand {
    pub fn new(
        customer_info: CreateOrderCommandCustomerInfo,
        items: Vec<CreateOrderCommandItem>,
        shipping_method_id: String,
        payment_method_id: String,
        shipping_address: CreateOrderCommandShippingAddress,
    ) -> Self {
        Self {
            customer_info,
            items,
            shipping_method_id,
            payment_method_id,
            shipping_address,
        }
    }
}

File: app/backend-ts/src/application/commands/models/create-order.command.ts
export class CreateOrderCommandItem {
  constructor(
    public readonly skuId: string,
    public readonly quantity: number,
  ) {}
}

export class CreateOrderCommandCustomerInfo {
  constructor(
    public readonly firstName: string,
    public readonly lastName: string,
    public readonly email: string,
    public readonly phone: string,
  ) {}
}

export class CreateOrderCommandShippingAddress {
  constructor(
    public readonly postalCode: string,
    public readonly prefecture: string,
    public readonly city: string,
    public readonly streetAddress: string,
    public readonly building?: string,
  ) {}
}

export class CreateOrderCommand {
  constructor(
    public readonly items: CreateOrderCommandItem[],
    public readonly customerInfo: CreateOrderCommandCustomerInfo,
    public readonly shippingAddress: CreateOrderCommandShippingAddress,
    public readonly shippingMethodId: string,
    public readonly paymentMethodId: string,
  ) {}
}
